scripts/qa/docs.sh

```diff
@@ -27,12 +25,19 @@ if [ ! -r docs/config.md ]; then
 	exit 2
 fi
 scripts/dev/config-to-doc.sh > $tmp
-diff docs/config.md $tmp
+if ! diff docs/config.md $tmp; then
+  echo "docs/config.md does not match output of scripts/dev/config-to-doc.sh"
+  rm $tmp
+  exit 2
+fi
+
+echo "checking metrics.md"
+go get github.com/Dieterbe/metrics2docs
+metrics2docs . > $tmp
+diff docs/metrics.md $tmp
 ret=$?
```

Comment:
For consistency I would either use the `ret=$?` every time you want to check the exit code, or use the `if ! <cmd>` every time, but not mix them within a short script.

Reply:
if we introduce 2 separate exit statements, then we also need to do the cleanup in both places.
seems simpler to do 1 cleanup followed by 1 exit, it warrants the extra ret assignment but that's just one line.

i see your point though. any exit code > 0 will do and i think we can argue both ways. I don't see a strong reason to prefer either approach over the other, so i suggest to keep the path of least work and keep it as is.